File: questions/library/go/helper/matrix.go
package helper

import "fmt"

func MatrixShow(matrix [][]int) (res string) {
	res = "["
	for rowNum, row := range matrix {
		res += "["
		for index, entry := range row {
			if index == 0 {
				res += fmt.Sprintf("%d", entry)
			} else {
				res += fmt.Sprintf(", %d", entry)
			}
		}
		if rowNum != len(matrix)-1 {
			res += "], "
		} else {
			res += "]"
		}
	}
	res += "]"
	return res
}

File: questions/library/typescript/matrix.ts
export default function (matrix: number[][]) {
  let res = "[";
  for (let rowNum = 0; rowNum < matrix.length; rowNum++) {
    res += "["
    for (let colNum = 0; colNum < matrix[rowNum].length; colNum++) {
      if (colNum == 0) {
        res += matrix[rowNum][colNum].toString();
      } else {
        res += ", " + matrix[rowNum][colNum].toString();
      }
    }
    if (rowNum != matrix.length - 1) {
      res += "], ";
    } else {
      res += "]";
    }
  }
  res += "]"
  return res;
}

